NonTerminals.cs

```diff
@@ -0,0 +1,162 @@
+                             firstNoVariantRecDependent = Production.TypeDeclarationTail,
+                             lastNoVariantRecDependent = Production.TypeDeclarationTail,
+                             firstOptionDependent = firstConstParamDependent,
+                             lastOptionDependent = lastNoVariantRecDependent;
+    public static int alternateSetOffset = (int)lastOptionDependent - (int)firstOptionDependent + 1;                         
+
+    /* --------------------------------------------------------------------------
+     * method Count() -- Returns the number of productions
+     * ----------------------------------------------------------------------- */ //SAM CHANGE
+
+    uint Count() {
+        uint total = 0;
+        foreach (String s in Enum.GetNames(typeof(Production))) {
+            total++;
+        }
+        return total;
+    } /* end Count */
+
+
+    /* --------------------------------------------------------------------------
+     * method IsOptionDependent(p)
+     * --------------------------------------------------------------------------
+     * Returns true if p is dependent on any compiler option, else false.
+     * ----------------------------------------------------------------------- */ //SAM CHANGE
+
+    bool IsOptionDependent(Production p) {
+        return p >= firstOptionDependent;
+    } /* end IsOptionDependent */
+
+
+    /* --------------------------------------------------------------------------
+     * method IsConstParamDependent(p)
+     * --------------------------------------------------------------------------
+     * Returns true if p is dependent on CONST parameter option, else false.
+     * ----------------------------------------------------------------------- */ //SAM CHANGE
+
+    bool IsConstParamDependent(Production p)
+    {
+        return p >= firstConstParamDependent && p <= lastConstParamDependent;
+    } /* end IsConstParamDependent */
+
+
+    /* --------------------------------------------------------------------------
+     * method IsVariantRecordDependent(p)
+     * --------------------------------------------------------------------------
+     * Returns true if p is dependent on variant record type option, else false.
+     * ----------------------------------------------------------------------- */ //SAM CHANGE
+
+    bool IsVariantRecordDependent(Production p) {
+        return p >= firstNoVariantRecDependent && p <= lastNoVariantRecDependent;
+    }
+
+    /* --------------------------------------------------------------------------
+     * method FIRST(p)
+     * --------------------------------------------------------------------------
+     * Returns a tokenset with the FIRST set of production p.
+     * ----------------------------------------------------------------------- */
+
+    TokenSet FIRST(Production p) {
+        TokenSet tokenset = null;
+
+        if (IsConstParamDependent(p))
+        {
+            p += alternateSetOffset;

```

Comment:
Is there a particular reason you like Convert.ToUInt32(p) over (uint)p?
Should run a find & replace and change all of the code to one or the other?